sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SyntaxTrackers/InvocationTracker.cs

```diff
@@ -62,59 +62,63 @@ void TrackInvocationExpression(SyntaxNodeAnalysisContext c)
 
             bool IsTrackedMethod(SyntaxNode invocation, SemanticModel semanticModel)
             {
-                var identifier = GetIdentifier(invocation);
-                if (identifier == null)
+                var methodName = GetMethodName(invocation);
+                if (methodName == null)
                 {
                     return false;
                 }
 
-                var conditionContext = new InvocationContext(invocation, identifier, semanticModel);
+                var conditionContext = new InvocationContext(invocation, methodName, semanticModel);
                 return conditions.All(c => c(conditionContext));
             }
         }
 
         #region Syntax-level standard conditions
 
-        public abstract InvocationCondition MatchSimpleNames(params MethodSignature[] methods);
-
         public InvocationCondition FirstParameterIsConstant() =>
-            ParameterAtIndexIsConstant(0);
-
-        public abstract InvocationCondition MethodNameIs(string methodName);
+            ArgumentAtIndexIsConstant(0);
 
-        public abstract InvocationCondition ParameterAtIndexIsConstant(int index);
+        public abstract InvocationCondition ArgumentAtIndexIsConstant(int index);
 
-        public abstract InvocationCondition ParameterAtIndexIsString(int index, string value);
+        public abstract InvocationCondition ArgumentAtIndexIsString(int index, string value);
```

Comment:
Maybe I am going to far but wouldn't it be better to have the method being something like `ArgumentAtIndexIsOfType<T>`?